ayushi/ayushi_postgresql_assignment/schema.sql

```diff
@@ -0,0 +1,108 @@
+CREATE DATABASE schoolDB;
+CREATE SCHEMA SchoolSchema;
+SET search_path TO SchoolSchema;
+CREATE TYPE address AS (
+	city VARCHAR(30),
+	street VARCHAR(40)
+);
+CREATE DOMAIN gender VARCHAR(15) 
+	CHECK (VALUE IN ('male','female','transgender'));
+
+CREATE TABLE IF NOT EXISTS students (
+	student_id INT PRIMARY KEY,
+	student_name VARCHAR(40) NOT NULL,
+	student_dob DATE NOT NULL,
+	student_address address NOT NULL,
+	student_gender gender NOT NULL,
+	student_phone VARCHAR(10) NOT NULL
+);
+CREATE TABLE IF NOT EXISTS teachers (
+	teacher_id INT PRIMARY KEY,
+	teacher_name VARCHAR(40) NOT NULL,
+	teacher_dob DATE NOT NULL,
+	teacher_address address NOT NULL,
+	teacher_gender gender NOT NULL,
+	teacher_phone VARCHAR(10) NOT NULL
+);
+CREATE TABLE IF NOT EXISTS subjects (
+	subject_id INT PRIMARY KEY,
+	subject_name VARCHAR(40) NOT NULL
+);
+CREATE TABLE IF NOT EXISTS schedule (
+	class_id SERIAL PRIMARY KEY,
+	subject_id INT NOT NULL,
+	teacher_id INT NOT NULL
+);
+CREATE TABLE IF NOT EXISTS student_class (
+	student_id INT NOT NULL,
+	class_id INT NOT NULL,
+	PRIMARY KEY (student_id,class_id)
+);
+CREATE TABLE IF NOT EXISTS results (
+	student_id INT NOT NULL,
```

Comment:
using composite key would help here ???

Reply:
each row of  table is defined by student_id and class_id together. So, PK is their combination.